lib/commands/web.js

```diff
@@ -0,0 +1,189 @@
+import { retryInterval } from 'asyncbox';
+import _ from 'lodash';
+import { errors } from 'mobile-json-wire-protocol';
+import logger from '../logger';
+
+
+let commands = {}, helpers = {}, extensions = {};
+
+const ELEMENT_OFFSET = 5000;
+
+commands.setUrl = async function (url) {
+  logger.debug(`Attempting to set url '${url}'`);
+  if (!this.isWebContext()) {
+    // in the future, detect whether we have a UIWebView that we can use to
+    // make sense of this command. For now, and otherwise, it's a no-op
+    throw new errors.NotImplementedError();
+  } else {
+    // make sure to clear out any leftover web frames
+    this.curWebFrames = [];
+    await this.remote.navToUrl(url);
+  }
+};
+
+commands.getUrl = async function () {
+  if (!this.isWebContext()) {
+    throw new errors.NotImplementedError();
+  } else {
+    return await this.remote.execute('window.location.href');
+  }
+};
+
+commands.title = async function () {
+  if (!this.isWebContext()) {
+    throw new errors.NotImplementedError();
+  } else {
+    return await this.executeAtom('title', []);
+  }
+};
+
+commands.findWebElementOrElements = async function (strategy, selector, ctx, many) {
+  let atomsElement = this.getAtomsElement(ctx);
+  let element;
+  let doFind = async () => {
+    element = await this.executeAtom(`find_element${many ? 's' : ''}`, [strategy, selector, atomsElement]);
+    return !_.isNull(element);
+  };
+  await this.implicitWaitForCondition(doFind);
+
+  if (many) {
+    return element;
+  } else {
+    if (!element || _.size(element) === 0) {
+      throw new errors.NoSuchElementError();
+    }
+    return element;
+  }
+};
+
+commands.click = async function (elementId) {
+  if (this.isWebContext()) {
+    let atomsElement = this.useAtomsElement(elementId);
+    return await this.executeAtom('click', [atomsElement]);
+  } else {
```

Comment:
basically just look anywhere you have an `else` in the whole project and chances are you can remove it :-)